### ISSUE TITLE
Click'N'Load: SSL: UNKNOWN_PROTOCOL

### ISSUE THREAD

**Issue by lugge86** [pyload/pyload#2560]
I set up a Raspberry-Pi2 with fresh Raspbian image (current version).
I downloaded pyload 0.4.9 from source and "installed" it in ~/pyload.
I replaced the UpdateManager.py file as I've read in some other topic around here.

I then started and configured pyload py invoking "python pyLoadCore.py"
Core is now up and running.
I can connect with CLI.
I can also connect with WebUI on my main PC.
I added some accounts and tested downloads. Works fine.

Now I wanted to use ClicknLoad.
Because pyload is running on my PI2, I installed a port-forward via SSH-tunnel on my main PC:
`ssh -L 127.0.0.1:9666:192.168.178.56:9666  pi@192.168.178.56 -N`

When I click a ClickNLoad link in browser on my main PC, pyLoadCore prints some text, thus, port forwarding seems to be working.
However, pyload does not add the links to queue but prints an error message (taken from log file):
```
30.07.2016 09:42:32 INFO      ADDON ClickNLoad: Proxy listening on 0.0.0.0:9666
30.07.2016 09:42:54 ERROR     ADDON ClickNLoad: [SSL: UNKNOWN_PROTOCOL] unknown protocol (_ssl.c:581)
30.07.2016 09:48:01 ERROR     ADDON ClickNLoad: [SSL: UNKNOWN_PROTOCOL] unknown protocol (_ssl.c:581)
30.07.2016 10:10:18 ERROR     ADDON ClickNLoad: [SSL: UNKNOWN_PROTOCOL] unknown protocol (_ssl.c:581)
```

When I check console output of the running pyloadcore, I get a little more information:
```
30.07.2016 10:12:32 INFO      ADDON ClickNLoad: Proxy listening on 0.0.0.0:9666
127.0.0.1 - - [30/Jul/2016 10:12:54] code 400, message Bad request syntax ('\x16\x03\x01\x02\x00\x01\x00\x01\xfc\x03\x03\x91\xf0\x08\xdf\x1d\xff\x81s\xb0\xbfh\x16\xf1\xa3\x98\x02\x02U\xe9\x963\xe0v`\xd0Iv\xf4\xd38\x02\xe6\x00\x00z\xc00\xc0,\xc02\xc0.\xc0/\xc0+\xc01\xc0-\x00\xa3\x00\x9f\x00\xa2\x00\x9e\xc0(\xc0$\xc0\x14\xc0')
30.07.2016 10:12:54 ERROR     ADDON ClickNLoad: [SSL: UNKNOWN_PROTOCOL] unknown protocol (_ssl.c:581)
```

Please note the additional line. (code 400).
Now, I'm not sure if this is a pyload-related issue or if the ClickNLoad link just contains bad data.
However, I tried multiple links from different sites.

Anyone encountered such issue?
Any work-around?

**Comment by lugge86** [comment]
I just tested with the "threaded" server instead of "builtin".
Seems to work fine using the same ClickNLoad links as before.
When using "builtin" again, the same errors occure.

Thus, I have to assume the issue is not related to the link or to my SSH tunnel but is indeed a bug. (In the "builtin" server's SSL handler?)

**Comment by GammaC0de** [comment]
:+1: 

Closing..

**Comment by lugge86** [comment]
Why is this closed?
I just confirmed that this issue is likely a bug in the builtin-server.

**Comment by GammaC0de** [comment]
Because:
1) You said that you solved your problem.
2) "builtin" server does not support SSL (probably only "threaded" supports this)

**Comment by lugge86** [comment]
Yeah, well, kind of "solved". I just stumbled over a workaround which works for me (using threaded server).

But OK, when you say SSL is not supported by builtin, there's no bug (it's rather a feature I guess...).
However, such stuff should be documented somewhere. It was by pure accident that I switched the server and "solved" the issue.
Have a look at the error log in my first post, the messages are strange and missleading. I would call that an issue, even if is known that builtin does not support SSL.

However, it's working for me now, so I'm fine.

**Comment by GammaC0de** [comment]
pyLoad was written by many developers over the years and there are some parts that are poorly documented.
Also, pyload.org site and forums where lost recently and so much documentation lost with it.

I did not remember the fact about "builtin" and SSL,
When you said that you have solved it by switching to "threaded", only then that reminded me seeing this when running pyLoad's setup (see `setup.py`)

I did take a look at your log before but it says the error comes from `ssl.c` (not even .py!) which is part of python and not part of pyLoad. so I suspected that you have issues with your python installation.
